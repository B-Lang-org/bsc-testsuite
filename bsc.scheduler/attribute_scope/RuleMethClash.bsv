interface Ifc;
   method Action incr();
endinterface

(* synthesize *)
module sysRuleMethClash (Ifc);
   Reg#(UInt#(16)) rg <- mkReg(0);

   rule incr;
      rg <= rg + 1;
   endrule

   Ifc i = interface Ifc;
              method Action incr();
                 rg <= rg + 2;
              endmethod
           endinterface;

   return i;
endmodule

