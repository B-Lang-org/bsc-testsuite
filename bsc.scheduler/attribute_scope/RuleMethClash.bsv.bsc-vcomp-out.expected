checking package dependencies
compiling RuleMethClash.bsv
code generation for sysRuleMethClash starts
Warning: "RuleMethClash.bsv", line 6, column 8: (G0010)
  Rule "incr" was treated as more urgent than "incr". Conflicts:
    "incr" cannot fire before "incr": calls to rg.write vs. rg.read
    "incr" cannot fire before "incr": calls to rg.write vs. rg.read
Verilog file created: sysRuleMethClash.v
All packages are up to date.
